### ISSUE TITLE
Akka-streams: Memory leak due to InboundTcpStreamActor not terminating on some errors.

### ISSUE THREAD

**Issue by nodefactory-bk** [akka/akka#19012]
Hi!

There seems to be an issue in TcpStreamActor where it will not terminate properly if an error happens while waiting for a WriteAck.

What happens is that a write is queued, some error happens calling the error() in the tcpOutputs singleton. However since a write is queued the lastWriteAcked var is false.
So tryShutdown will not terminate the actor since isFlushed returns false.

This seems to happen both with version 1.0 and 2.0-M1.

This trivial patch seems to fix the issue:
```
--- a/akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala
+++ b/akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala
@@ -137,6 +137,7 @@ private[akka] abstract class TcpStreamActor(val settings: Ac
torMaterializerSetti
     override def error(e: Throwable): Unit = {
       if (!closed && initialized) connection ! Abort
       closed = true
+      lastWriteAcked = true
     }
 
     override def complete(): Unit = {
```

it just ensures that lastWriteAcked is set to true if an error occured.
Don't know if this is the preferred fix, but it WorksForMe<tm> =)

**Comment by drewhk** [comment]
Thanks for reporting! I think this is no longer relevant though because of the rewrite in: https://github.com/akka/akka/pull/18973

which does not rely on that old actor infrastructure. We should verify it, but I am pretty sure this is no longer an issue.

**Comment by 2beaucoup** [comment]
Can be closed?

**Comment by nodefactory-bk** [comment]
Think it was fixed in #18973 as stated above. hasn't affected me since then.

**Comment by ktoso** [comment]
Very glad to hear that, thanks for confirming!
